model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/UserCacheSession.java

```diff
@@ -338,7 +338,7 @@ protected UserModel validateCache(RealmModel realm, CachedUser cached) {
     protected UserModel cacheUser(RealmModel realm, UserModel delegate, Long revision) {
         int notBefore = getDelegate().getNotBeforeOfUser(realm, delegate);
 
-        if (Profile.isFeatureEnabled(Profile.Feature.ORGANIZATION)) {
+        if (Profile.isFeatureEnabled(Profile.Feature.ORGANIZATION) && realm.isOrganizationsEnabled()) {
```

Comment:
We can't check it here because the user will be readonly when orgs is disabled and the user is a "managed" user (a user that was created in the context of an org).

Reply:
Hmm, but this method is checking whether the organization itself is disabled, not organizations in general.

It seems this method is not checking realm config, only org config or am I missing something?
https://github.com/keycloak/keycloak/blob/8d314a69a01f0909cc94ebbf5d04ecab8de69674/model/storage-private/src/main/java/org/keycloak/storage/UserStorageManager.java#L936-L943

Reply:
Behind the scenes, the call to `organizationProvider.isEnabled` is checking if org is enabled at the realm level.

Reply:
Ok. Can we then change it something like the following? It would avoid looking members by id if orgProvider is disabled. 
```
private boolean isOrganizationDisabled(KeycloakSession session, UserModel delegate) {
        // check if provider is enabled and user is managed member of a disabled organization OR provider is disabled and user is managed member
        OrganizationProvider organizationProvider = session.getProvider(OrganizationProvider.class);

        if (!organizationProvider.isEnabled()) {
            return true;
        }
 
        return organizationProvider.getByMember(delegate)
                .anyMatch((org) -> (organizationProvider.isEnabled() && org.isManaged(delegate) && !org.isEnabled()) ||
                        (!organizationProvider.isEnabled() && org.isManaged(delegate)));
    }
```

Reply:
I don't think so. As I said, if org is disabled (at the realm) and the user is a managed member, we need to return false.

Reply:
We could at least avoid fetching users to check for memberships if there are no orgs in the realm.. so in addition to checking if the feature is enabled, we could do 
```
OrganizationProvider organizationProvider = session.getProvider(OrganizationProvider.class);
if(Profile.isFeatureEnabled(Profile.Feature.ORGANIZATION) && organizationProvider.count > 0) {
 ....
 }
```